searx/engines/digg.py

```diff
@@ -1,67 +1,66 @@
 """
  Digg (News, Social media)
 
- @website     https://digg.com/
+ @website     https://digg.com
  @provide-api no
 
  @using-api   no
  @results     HTML (using search portal)
  @stable      no (HTML can change)
  @parse       url, title, content, publishedDate, thumbnail
 """
+# pylint: disable=missing-function-docstring
 
-import random
-import string
 from json import loads
 from urllib.parse import urlencode
 from datetime import datetime
 
+from lxml import html
+
 # engine dependent config
 categories = ['news', 'social media']
 paging = True
+base_url = 'https://digg.com'
 
 # search-url
-base_url = 'https://digg.com/'
-search_url = base_url + 'api/search/?{query}&from={position}&size=20&format=html'
-
-# specific xpath variables
-results_xpath = '//article'
-link_xpath = './/small[@class="time"]//a'
-title_xpath = './/h2//a//text()'
-content_xpath = './/p//text()'
-pubdate_xpath = './/time'
-
-digg_cookie_chars = string.ascii_uppercase + string.ascii_lowercase +\
-    string.digits + "+_"
-
+search_url = base_url + (
+    '/api/search/'
+    '?{query}'
+    '&from={position}'
+    '&size=20'
+    '&format=html'
+)
 
-# do search-request
 def request(query, params):
     offset = (params['pageno'] - 1) * 20
-    params['url'] = search_url.format(position=offset,
-                                      query=urlencode({'q': query}))
-    params['cookies']['frontend.auid'] = ''.join(random.choice(
-        digg_cookie_chars) for _ in range(22))
+    params['url'] = search_url.format(
+        query = urlencode({'q': query}),
+        position = offset,
+    )
     return params
 
-
-# get response from search-request
 def response(resp):
     results = []
 
-    search_result = loads(resp.text)
-
     # parse results
-    for result in search_result['mapped']:
+    for result in loads(resp.text)['mapped']:
+
+        # strip html tags and superfluous quotation marks from content
+        content = html.document_fromstring(
+            result['excerpt']
+        ).text_content().strip('\'"\\n')
```

Comment:
The PR LGTM except this little tiny thing : I guess `'\'"\\n'` should be `'\'"\n'` otherwise it removes `n` and `\` ?

![image](https://user-images.githubusercontent.com/1594191/99943245-45c49e80-2d71-11eb-9000-dd134c541087.png)

Reply:
I found the two characters `\` and `n` at the end of some hits / deleting a common `n` at the end is in deed a bug.

I am very sorry, I do not have time to fix and rebase these days.